test/metabase/models/params/chain_filter_test.clj

```diff
@@ -670,3 +671,25 @@
           (testing "`true` if the limit is less than the number of values the field has"
             (is (= true
                    (:has_more_values (chain-filter venues.latitude {venues.price 4} :limit 1))))))))))
+
+(deftest chain-filter-inactive-test
+  (testing "Inactive fields are not used to generate joins"
+    (mt/dataset avian-singles
+      (mt/$ids nil
+        (testing "receiver_id is active and should be used for the join"
+          (memoize/memo-clear! @#'chain-filter/database-fk-relationships)
+          (memoize/memo-clear! @#'chain-filter/find-joins)
```

Comment:
A `without-memoization` macro that hides these details could be nice. 

Implementation wise it'd be nice if it would be nice for something like this to also avoids populating the cache for any internal calls too, then you'd only need to wrap this whole test once.

To simplify the macro I guess you could pass the list of functions, then it's fairly straightfoward.

```clojure
(without-memoization [#'chain-filter/database-fk-relationships
                      #'chain-filter/find-joins]
  ...)
```

```clojure
(defmacro without-memoization [fns & body]
   `(do (doseq [f fns] (memoize/memo-clear! f))
        (mt/with-dynamic-redefs [~@(map wrap-them-so-they-clear-afterwards fns)]
          ~@body))
```

Reply:
Eh, it won't be doing `without-memoization`, so it'll be misleading somewhat. But it now makes me think if just `with-redefs` is going to be easier to test with. :)

Reply:
Am I missing something - I thought you were just using the clears to bypass any pre-existing cache?

Where possible try to use `mt/with-dynamic-redefs` instead, it's the threadsafe version 😄